api/src/main/java/uk/gov/hmcts/payment/api/controllers/payment/ReconciliationApiController.java

```diff
@@ -0,0 +1,211 @@
+    })
+    @GetMapping(value = "/payments")
+    @PaymentExternalAPI
+    public PaymentsResponse retrievePayments(@RequestParam(name = "start_date", required = false) Optional<String> startDateTimeString,
+                                             @RequestParam(name = "end_date", required = false) Optional<String> endDateTimeString,
+                                             @RequestParam(name = "payment_method", required = false) Optional<String> paymentMethodType,
+                                             @RequestParam(name = "service_name", required = false) Optional<String> serviceType,
+                                             @RequestParam(name = "ccd_case_number", required = false) String ccdCaseNumber,
+                                             @RequestParam(name = "pba_number", required = false) String pbaNumber
+    ) {
+
+        if (!ff4j.check("payment-search")) {
+            throw new PaymentException("Payment search feature is not available for usage.");
+        }
+
+        validator.validate(paymentMethodType, serviceType, startDateTimeString, endDateTimeString);
+
+        Date fromDateTime = Optional.ofNullable(startDateTimeString.map(formatter::parseLocalDateTime).orElse(null))
+            .map(LocalDateTime::toDate)
+            .orElse(null);
+
+        Date toDateTime = Optional.ofNullable(endDateTimeString.map(formatter::parseLocalDateTime).orElse(null))
+            .map(s -> fromDateTime != null && s.getHourOfDay() == 0 ? s.plusDays(1).minusSeconds(1).toDate() : s.toDate())
+            .orElse(null);
+
+        List<PaymentFeeLink> paymentFeeLinks = paymentService
+            .search(
+                PaymentSearchCriteria
+                    .searchCriteriaWith()
+                    .startDate(fromDateTime)
+                    .endDate(toDateTime)
+                    .ccdCaseNumber(ccdCaseNumber)
+                    .pbaNumber(pbaNumber)
+                    .paymentMethod(paymentMethodType.map(value -> PaymentMethodType.valueOf(value.toUpperCase()).getType()).orElse(null))
+                    .serviceType(serviceType.map(value -> Service.valueOf(value.toUpperCase()).getName()).orElse(null))
+                    .build()
+            );
+
+        final List<PaymentDto> paymentDtos = new ArrayList<>();
+        LOG.info("No of paymentFeeLinks retrieved for Liberata Pull : {}", paymentFeeLinks.size());
+        for (final PaymentFeeLink paymentFeeLink: paymentFeeLinks) {
+            populatePaymentDtos(paymentDtos, paymentFeeLink);
+        }
+        return new PaymentsResponse(paymentDtos);
+    }
+
+    @ApiOperation(value = "Update payment status by payment reference", notes = "Update payment status by payment reference")
+    @ApiResponses(value = {
+        @ApiResponse(code = 204, message = "No content"),
+        @ApiResponse(code = 404, message = "Payment not found")
+    })
+    @PaymentExternalAPI
+    @PatchMapping("/payments/{reference}/status/{status}")
```

Comment:
I understand that we are moving **/payments** which is the end point used by liberata to pull the data to ReconciliationApiController. But why we need to move **/payments/{reference}/status/{status}** to ReconciliationApiController?

Reply:
This is another external API which might get noticed in future ITHC reviews and we need to refactor, so we have moved to separate external API Controller.